public/app/plugins/datasource/prometheus/add_label_to_query.ts

```diff
@@ -15,8 +17,12 @@ const builtInWords = [
   .join('|')
   .split('|');
 
-const metricNameRegexp = /([A-Za-z:][\w:]*)\b(?![\(\]{=!",])/g;
-const selectorRegexp = /{([^{]*)}/g;
+// We want to extract all possible metrics and also keywords
+const metricsAndKeywordsRegexp = /([A-Za-z:][\w:]*)\b(?![\]{=!",])/g;
+// Safari currently doesn't support negative lookbehind. When it does, we should refactor this.
+// We are creating 2 matching groups. (\$) is for the Grafana's variables such as ${__rate_s}. We want to ignore
+// ${__rate_s} and not add variable to it.
+const selectorRegexp = /(\$)?{([^{]*)}/g;
```

Comment:
This could be solved with negative lookbehind `(?<!\$){([^{]*)}` and we would receive only selectors that don't start with `$`. Unfortunately Safari doesn't support that https://caniuse.com/?search=lookbehind :( . I have followed this https://stackoverflow.com/questions/35142364/regex-negative-lookbehind-not-valid-in-javascript as a way how you can implement global negative-lookbehind behaviour.